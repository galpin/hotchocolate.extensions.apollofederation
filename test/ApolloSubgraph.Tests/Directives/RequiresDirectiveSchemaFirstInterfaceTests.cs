using System.Threading.Tasks;
using HotChocolate.Types;
using Microsoft.Extensions.DependencyInjection;
using Xunit;

namespace HotChocolate.Extensions.ApolloSubgraph.Directives;

public class RequiresDirectiveSchemaFirstInterfaceTests
{
    [Fact]
    public async Task When_requires_is_specified_on_object()
    {
        var schema = await Test.BuildSchemaAsync(builder =>
        {
            builder.AddDocumentFromString(@"
                interface IReview @key(fields: ""id"") {
                     id: Int
                     product: Product @requires(fields: ""id"")
                }

                type Review implements IReview @key(fields: ""id"") {
                     id: Int
                     product: Product @requires(fields: ""id"")
                }

                type Product @key(fields: ""upc"") {
                     name: String
                }

                type Query
            ");
        });

        var sut = schema.GetType<InterfaceType>("IReview");

        Assert.Collection(
            sut.Fields["product"].Directives,
            x => AssertEx.Directive(x, "requires", ("fields", "\"id\"")));
        await schema.QuerySdlAndMatchSnapshotAsync();
    }
}