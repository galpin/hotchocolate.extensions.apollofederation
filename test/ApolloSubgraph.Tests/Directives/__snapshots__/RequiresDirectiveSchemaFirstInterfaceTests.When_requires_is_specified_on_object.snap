{
  "data": {
    "_service": {
      "sdl": "interface IReview @key(fields: \u0022id\u0022) {\n  id: Int\n  product: Product @requires(fields: \u0022id\u0022)\n}\n\ntype Product @key(fields: \u0022upc\u0022) {\n  name: String\n}\n\ntype Review implements IReview @key(fields: \u0022id\u0022) {\n  id: Int\n  product: Product @requires(fields: \u0022id\u0022)\n}"
    }
  }
}
